### ISSUE TITLE
Closes socket when a client tries fetching any of the test actors

### ISSUE THREAD

**Issue by grishka** [go-fed/testsuite#2]
Not sure if a bug or a server configuration issue, but...

```
$ curl -v -H "Accept: activity+json" https://test.activitypub.dev/tests/evaluate/4037200794235010051/alex
*   Trying 104.131.5.48...
* TCP_NODELAY set
* Connected to test.activitypub.dev (104.131.5.48) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: CN=test.activitypub.dev
*  start date: Jul 16 14:40:26 2020 GMT
*  expire date: Oct 14 14:40:26 2020 GMT
*  subjectAltName: host "test.activitypub.dev" matched cert's "test.activitypub.dev"
*  issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
*  SSL certificate verify ok.
> GET /tests/evaluate/4037200794235010051/alex HTTP/1.1
> Host: test.activitypub.dev
> User-Agent: curl/7.54.0
> Accept: activity+json
> 
* TLSv1.2 (IN), TLS alert, Client hello (1):
* Empty reply from server
* Connection #0 to host test.activitypub.dev left intact
curl: (52) Empty reply from server
```

This happens regardless of the accept header.

**Comment by cjslep** [comment]
Each test also has a 15 minute TTL, was this within that time period?

I see a panic that might indicate a bug as well.
```
http: panic serving 188.227.48.38:64006: runtime error: invalid memory address or nil pointer dereference
goroutine 274 [running]:
net/http.(*conn).serve.func1(0xc0005f8320)
        /usr/local/go/src/net/http/server.go:1767 +0x139
panic(0x15c3520, 0x30ee090)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
net/url.(*URL).String(0x0, 0x161ae00, 0x0)
        /usr/local/go/src/net/url/url.go:775 +0x4a
github.com/go-fed/testsuite/server.(*Recorder).Add(0xc0002bb040, 0x1729378, 0x15, 0xc00016b7a8, 0x7, 0x7)
        /Users/cjslep/go/src/github.com/go-fed/testsuite/server/recorder.go:54 +0x7d4
github.com/go-fed/testsuite/server.(*TestRunner).LogPubHandlerFunc(0xc00009a420, 0x2604460, 0xc0002003f0, 0xc0000d0400, 0xc0000d0400, 0x0, 0x0, 0x0, 0x0, 0x25f9d60, ...)
        /Users/cjslep/go/src/github.com/go-fed/testsuite/server/testrunner.go:606 +0x199
github.com/go-fed/testsuite/server.(*Actor).PubHandlerFunc(0xc0002bb060, 0x2604460, 0xc0002003c0, 0x2602560, 0xc0006520e0, 0xc0000d0400, 0x1, 0xc000204570, 0x28)
        /Users/cjslep/go/src/github.com/go-fed/testsuite/server/actor.go:227 +0x233
github.com/go-fed/testsuite/server.(*TestServer).HandleWeb(0xc00008e140, 0x2604460, 0xc0002003c0, 0x2602560, 0xc0006520e0, 0xc0000d0400)
        /Users/cjslep/go/src/github.com/go-fed/testsuite/server/testserver.go:248 +0x5c9
github.com/go-fed/testsuite/server.(*WebServer).testRequestHandler(0xc0000edbf0, 0x2602560, 0xc0006520e0, 0xc0000d0400)
        /Users/cjslep/go/src/github.com/go-fed/testsuite/server/webserver.go:131 +0xff
net/http.HandlerFunc.ServeHTTP(0xc000055df0, 0x2602560, 0xc0006520e0, 0xc0000d0400)
        /usr/local/go/src/net/http/server.go:2007 +0x44
net/http.(*ServeMux).ServeHTTP(0xc0000f2140, 0x2602560, 0xc0006520e0, 0xc0000d0400)
        /usr/local/go/src/net/http/server.go:2387 +0x1bd
net/http.serverHandler.ServeHTTP(0xc0000201c0, 0x2602560, 0xc0006520e0, 0xc0000d0400)
        /usr/local/go/src/net/http/server.go:2802 +0xa4
net/http.(*conn).serve(0xc0005f8320, 0x26043a0, 0xc000062440)
        /usr/local/go/src/net/http/server.go:1890 +0x875
created by net/http.(*Server).Serve
```

**Comment by grishka** [comment]
> Each test also has a 15 minute TTL, was this within that time period?

Yes, I initially tried using my "external interaction" UI to follow test instructions.

> 188.227.48.38

Yes that's my IP :)

**Comment by cjslep** [comment]
Deleted my previous post, I don't want to share your IP. Sorry for accidentally posting it.

Seems like it is a bug triggered here:
```
/Users/cjslep/go/src/github.com/go-fed/testsuite/server/testrunner.go:606
```

When HTTP Signatures is not used. Due to the nature of `nil` and interface type switches, the method used to log the passed-in data attempts to dereference a nil when the request is not using HTTP Signatures. So it is a bug in the server.

**Comment by cjslep** [comment]
Should now be fixed by ff03961ef8db0e133e0c6ccc05a87ad799b818f2

And be hosted at the testsuite URL.
